pkg/cli/cmd/run/run.go

```diff
@@ -215,3 +258,29 @@ func (r *Runner) displayPortforwardMessages(status <-chan portforward.StatusMess
 		fmt.Printf("%s %s [port-forward] %s from localhost:%d -> ::%d\n", regular.Sprint(message.ReplicaName), bold.Sprint(message.ContainerName), message.Kind, message.LocalPort, message.RemotePort)
 	}
 }
+
+// dashboardDeploymentExists checks if a dashboard deployment exists in the given Kubernetes context.
+func dashboardDeploymentExists(ctx context.Context, kubernetesClient k8sclient.Interface, kubeContext string, dashboardLabelSelector labels.Selector) bool {
+	// If a kubernetes client is provided, use it. Otherwise, create a new client.
+	// This helps with testing.
+	if kubernetesClient == nil {
+		var err error
+		kubernetesClient, _, err = kubernetes.NewClientset(kubeContext)
```

Comment:
Is this used for testing? If kubeContext doesn't exist in local machine, you will get the failure. Do not use this code. instead, we need to pass fake client to this. Can you pass fake client to dashboardDeploymentExists in the test without having this if ?